client/src/features/session/startSessionOptionsSlice.ts

```diff
@@ -110,6 +113,12 @@ export const startSessionOptionsSlice = createSlice({
     setSessionClass: (state, action: PayloadAction<number>) => {
       state.sessionClass = action.payload;
     },
+    setSecretsList: (state, action: PayloadAction<SessionSecrets[]>) => {
+      state.secretsList = action.payload;
+    },
+    setSecretsPath: (state, action: PayloadAction<string>) => {
+      state.secretsPath = action.payload;
+    },
```

Comment:
I think it would be better to include a `toggleSecret` action as well. This would be something like:

```suggestion
    },
    toggleSecret(state, action: PayloadAction<SessionSecrets>) => {
          const secret = action.payload;
          const newSecretsList = state.secretsList.map((s) => s.name).includes(secret.name)) ?
            state.secretsList.filter((s) => s.name !== secret.name)
          ) :  [...state.secretsList, secret]));
       state.secretsList = newSecretsList
     },
```

Reply:
I'm not sure this is needed. It basically moves the logic of `setSecretsPath` to the slice, but it's used only in one place.
If you don't mind, I would prefer to keep the logic simple there (as with all the other reducers) and keep the toggle-like implementation in the container.